src/Symfony/Bundle/SecurityBundle/DependencyInjection/MainConfiguration.php

```diff
@@ -191,7 +191,12 @@ private function addFirewallsSection(ArrayNodeDefinition $rootNode, array $facto
         ;
 
         $firewallNodeBuilder
-            ->scalarNode('pattern')->end()
+            ->scalarNode('pattern')
+                ->beforeNormalization()
+                    ->ifTrue(fn ($v) => \is_array($v))
+                    ->then(fn ($v) => sprintf('(%s)', implode('|', $v)))
```

Comment:
Nit-pick: I think we don't want to capture that group.

```suggestion
                    ->then(fn ($v) => sprintf('(?:%s)', implode('|', $v)))
```

Reply:
Thanks, fixed